plugins/repository-hdfs/build.gradle

```diff
@@ -7,427 +7,385 @@
-        return fixture.portsFile.exists()
-      }
-      final List<String> miniHDFSArgs = []
-
-      // If it's a secure fixture, then depend on Kerberos Fixture and principals + add the krb5conf to the JVM options
-      if (name.startsWith('secure')) {
-        miniHDFSArgs.addAll(["--add-exports", "java.security.jgss/sun.security.krb5=ALL-UNNAMED"])
-        miniHDFSArgs.add("-Djava.security.krb5.conf=${krb5conf}")
-        miniHDFSArgs.add("-Dhdfs.config.port=" + getSecureNamenodePortForVersion(hadoopVer))
-      } else {
-        miniHDFSArgs.add("-Dhdfs.config.port=" + getNonSecureNamenodePortForVersion(hadoopVer))
-      }
-      // If it's an HA fixture, set a nameservice to use in the JVM options
-      if (name.startsWith('haHdfs') || name.startsWith('secureHaHdfs')) {
-        miniHDFSArgs.add("-Dha-nameservice=ha-hdfs")
-      }
-
-      // Common options
-      miniHDFSArgs.add('hdfs.MiniHDFS')
-      miniHDFSArgs.add(baseDir)
-
-      // If it's a secure fixture, then set the principal name and keytab locations to use for auth.
-      if (name.startsWith('secure')) {
-        miniHDFSArgs.add("hdfs/hdfs.build.elastic.co@${realm}")
-        miniHDFSArgs.add(
-          project(':test:fixtures:krb5kdc-fixture').ext.krb5Keytabs("hdfs", "hdfs_hdfs.build.elastic.co.keytab")
-        )
-      }
-
-      args miniHDFSArgs.toArray()
+
+    configurations.create("hdfs" + hadoopVersion + "Fixture")
+    dependencies.add("hdfs" + hadoopVersion + "Fixture", project(':test:fixtures:hdfs' + hadoopVersion + '-fixture'))
+
+    for (String fixtureName : ['hdfs' + hadoopVersion + 'Fixture', 'haHdfs' + hadoopVersion + 'Fixture', 'secureHdfs' + hadoopVersion + 'Fixture', 'secureHaHdfs' + hadoopVersion + 'Fixture']) {
+        project.tasks.register(fixtureName, org.elasticsearch.gradle.internal.test.AntFixture) {
+            executable = "${BuildParams.runtimeJavaHome}/bin/java"
+            dependsOn project.configurations.getByName("hdfs" + hadoopVer + "Fixture"), project.configurations.krb5Config, project.configurations.krb5Keytabs
+            env 'CLASSPATH', "${-> project.configurations.getByName("hdfs" + hadoopVer + "Fixture").asPath}"
+
+            maxWaitInSeconds 60
+            onlyIf { BuildParams.inFipsJvm == false }
+            waitCondition = { fixture, ant ->
+                // the hdfs.MiniHDFS fixture writes the ports file when
+                // it's ready, so we can just wait for the file to exist
+                return fixture.portsFile.exists()
+            }
+            final List<String> miniHDFSArgs = []
+
+            // If it's a secure fixture, then depend on Kerberos Fixture and principals + add the krb5conf to the JVM options
+            if (name.startsWith('secure')) {
+                miniHDFSArgs.addAll(["--add-exports", "java.security.jgss/sun.security.krb5=ALL-UNNAMED"])
+                miniHDFSArgs.add("-Djava.security.krb5.conf=${project.configurations.krb5Config.getSingleFile().getPath()}")
```

Comment:
Wouldn't this trigger some dependency resolution at task graph calculation time or is that fine in this case since the "dependencies" are just a folder.

Reply:
we have added `project.configurations.krb5Keytabs` explicitly as dependency via `dependsOn` to ensure the task dependencies are resolved correctly